Test/Microsoft.ApplicationInsights.Test/Shared/Extensibility/Implementation/Tracing/HeartbeatTests.cs

```diff
@@ -0,0 +1,357 @@
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatWithNonDefaultInterval()
+        {
+            TimeSpan nonDefaultInterval = TimeSpan.FromMilliseconds(10000);
+
+            using (var hbeat = new HeartbeatProvider() { HeartbeatInterval = nonDefaultInterval })
+            {
+                hbeat.Initialize(configuration: null);
+                Assert.AreEqual(nonDefaultInterval, hbeat.HeartbeatInterval);
+            }
+        }
+
+        [TestMethod]
+        public void InitializeHealthHeartbeatWithZeroIntervalRevertsToDefault()
+        {
+            using (var hbeat = new HeartbeatProvider() { HeartbeatInterval = TimeSpan.FromMilliseconds(0) })
+            {
+                hbeat.Initialize(configuration: null);
+                Assert.AreEqual(hbeat.HeartbeatInterval, HeartbeatProvider.DefaultHeartbeatInterval);
+            }
+        }
+
+        [TestMethod]
+        public void CanExtendHeartbeatPayload()
+        {
+
+            using (var hbeat = new HeartbeatProvider())
+            {
+                hbeat.Initialize(configuration: new TelemetryConfiguration());
+
+                try
+                {
+                    Assert.IsTrue(hbeat.AddHealthProperty("test01", "this is a value", true));
+                }
+                catch (Exception e)
+                {
+                    Assert.Fail(string.Format(CultureInfo.CurrentCulture, "Registration of a heartbeat payload provider throws exception '{0}", e.ToInvariantString()));
+                }
+            }
+        }
+
+        [TestMethod]
+        public void CanSetDelayBetweenHeartbeats()
+        {
+            TimeSpan userSetInterval = TimeSpan.FromMilliseconds(7252.0);
+
+            using (var hbeat = new HeartbeatProvider())
+            {
+                var config = new TelemetryConfiguration(string.Empty, new StubTelemetryChannel());
+                hbeat.Initialize(configuration: config);
+                Assert.AreNotEqual(userSetInterval, hbeat.HeartbeatInterval.TotalMilliseconds);
+
+                hbeat.HeartbeatInterval = userSetInterval;
+                hbeat.Initialize(configuration: config);
+                Assert.AreEqual(userSetInterval, hbeat.HeartbeatInterval);
+            }
+        }
+
+        [TestMethod]
+        public void DiagnosticsTelemetryModuleCreatesHeartbeatModule()
+        {
+            using (var diagModule = new DiagnosticsTelemetryModule())
+            {
+                diagModule.Initialize(new TelemetryConfiguration());
+                Assert.IsNotNull(diagModule.HeartbeatProvider);
+            }
+        }
+
+        [TestMethod]
+        public void HeartbeatPayloadContainsDataByDefault()
+        {
+            using (var hbeat = new HeartbeatProvider())
+            {
+                var config = new TelemetryConfiguration(string.Empty, new StubTelemetryChannel());
```

Comment:
StubTelemetryChannel - is it needed here? Its not used anywhere

Reply:
No, it's not used here at all. I probably went overboard using StubTelemetryChannel when I learned of its use is all. I can remove it, but does it make any difference?

Reply:
If not needed then please remove as it'll make it obvious what is being tested and avoid confusion about whether Stubs/Mocks have any role here at all.

Reply:
Good point. I'll make the change.